### ISSUE TITLE
Jetty 12: X-Powered-By header is added 2 times (if enabled)

### ISSUE THREAD

**Issue by uschindler** [eclipse/jetty.project#10309]
**Jetty version(s)**
Jetty 12

**Jetty Environment**
core

**Description**
If you enable `HttpConfiguration#setSendXPoweredBy(true)`, then Jetty returns the header 2 times (at least with HTTP/1.1):

```
> curl -I http://localhost:8081/
HTTP/1.1 200 OK
X-Powered-By: Jetty(12.0.0)
Date: Mon, 14 Aug 2023 18:38:26 GMT
Content-Type: text/plain;charset=utf-8
Content-Length: 46
X-Powered-By: Jetty(12.0.0)
```

**Comment by uschindler** [comment]
Simple reproducer:

```java
public class JettyTest {

  public static void main(String[] args) throws Exception {
    final var server=new Server();
    
    // setup connector
    final var http_config = new HttpConfiguration();
    final var cust = new ForwardedRequestCustomizer();
    http_config.addCustomizer(cust);
    http_config.setSendServerVersion(false);
    http_config.setSendXPoweredBy(true);
    final var connector = new ServerConnector(server, new HttpConnectionFactory(http_config));
    connector.setHost(System.getProperty("server.host", "127.0.0.1"));
    connector.setPort(Integer.getInteger("server.port", 8081));
    server.addConnector(connector);

    // startup:
    server.start();
    server.join();
  }
}
```

**Comment by joakime** [comment]
Opened PR #10310 to address this

**Comment by joakime** [comment]
Basic fix merged in PR #10310
